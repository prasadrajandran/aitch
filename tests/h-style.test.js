import prettier from 'prettier';
import { hStyle } from '../dist';

const testData = {
  'Parses a single block': {
    div: {
      fontSize: '10px',
    },
  },
  'Parses multiple blocks': {
    div: {
      fontSize: '10px',
    },
    p: {
      display: 'flex',
      flexDirection: 'column',
      fontSize: '10px',
      fontFamily: "'Bungee Shade', system-ui",
    },
  },
  'Parses block with multiple selectors': {
    'div, p': {
      fontSize: '10px',
      display: 'flex',
      flexDirection: 'column',
      fontFamily: "'Bungee Shade', system-ui",
    },
  },
  'Parses blocks under a media query': {
    '@media (height: 360px)': {
      div: {
        fontSize: '10px',
      },
      p: {
        display: 'flex',
        flexDirection: 'column',
        fontSize: '10px',
      },
    },
  },
  'Parses animation keyframes': {
    '@keyframes slideIn': {
      from: {
        transform: 'translateX(0%)',
      },
      to: {
        transform: 'translateX(100%)',
      },
    },
  },
};

const runAndFormat = (data) => prettier.format(hStyle(data), { parser: 'css' });

describe('hStyle()', () => {
  test('Expects an argument', () => {
    expect(() => hStyle()).toThrowError();
  });

  test('Parses empty object', () => {
    expect(runAndFormat({})).toMatchSnapshot();
  });

  for (const [testName, data] of Object.entries(testData)) {
    test(testName, () => {
      expect(runAndFormat(data)).toMatchSnapshot();
    });
  }
});
