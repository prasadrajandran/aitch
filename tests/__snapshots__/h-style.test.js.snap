// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hStyle() Parses a single block 1`] = `
"div {
  font-size: 10px;
}
"
`;

exports[`hStyle() Parses animation keyframes 1`] = `
"@keyframes slideIn {
  from {
    transform: translateX(0%);
  }
  to {
    transform: translateX(100%);
  }
}
"
`;

exports[`hStyle() Parses block with multiple selectors 1`] = `
"div,
p {
  font-size: 10px;
  display: flex;
  flex-direction: column;
  font-family: "Bungee Shade", system-ui;
}
"
`;

exports[`hStyle() Parses blocks under a media query 1`] = `
"@media (height: 360px) {
  div {
    font-size: 10px;
  }
  p {
    display: flex;
    flex-direction: column;
    font-size: 10px;
  }
}
"
`;

exports[`hStyle() Parses empty object 1`] = `""`;

exports[`hStyle() Parses multiple blocks 1`] = `
"div {
  font-size: 10px;
}
p {
  display: flex;
  flex-direction: column;
  font-size: 10px;
  font-family: "Bungee Shade", system-ui;
}
"
`;
